From d9a2d3c03a23473e622a62c5876a8d9b8047604a Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Wed, 19 May 2021 07:03:14 -0400
Subject: [PATCH 2/2] Revert "keystore: Block key attestation for Google Play
 Services"

This reverts commit 8db8879154b540eb5b3903325d0b0f5e58bf77f6.

  RTFM: Danny Lin wrote "use only ONE method, no need both"
---
 keystore/key_store_service.cpp       | 11 +++--------
 keystore/keystore_attestation_id.cpp |  6 ------
 2 files changed, 3 insertions(+), 14 deletions(-)

diff --git a/keystore/key_store_service.cpp b/keystore/key_store_service.cpp
index b1f1304..1b38643 100644
--- a/keystore/key_store_service.cpp
+++ b/keystore/key_store_service.cpp
@@ -49,7 +49,6 @@
 #include <keystore/keystore_return_types.h>
 
 #include <hardware/hw_auth_token.h>
-#include <hardware/keymaster_defs.h>
 
 namespace keystore {
 
@@ -121,13 +120,9 @@ KeyStoreServiceReturnCode updateParamsForAttestation(uid_t callingUid, Authoriza
 
     auto asn1_attestation_id_result = security::gather_attestation_application_id(callingUid);
     if (!asn1_attestation_id_result.isOk()) {
-        if (asn1_attestation_id_result.status() == KM_ERROR_UNIMPLEMENTED) {
-            return KeyStoreServiceReturnCode(KM_ERROR_UNIMPLEMENTED);
-        } else {
-            ALOGE("failed to gather attestation_id");
-            // Couldn't get attestation ID; just use an empty one rather than failing.
-            asn1_attestation_id_result = std::vector<uint8_t>();
-        }
+        ALOGE("failed to gather attestation_id");
+        // Couldn't get attestation ID; just use an empty one rather than failing.
+        asn1_attestation_id_result = std::vector<uint8_t>();
     }
     std::vector<uint8_t>& asn1_attestation_id = asn1_attestation_id_result;
 
diff --git a/keystore/keystore_attestation_id.cpp b/keystore/keystore_attestation_id.cpp
index 448a909..3d9e87e 100644
--- a/keystore/keystore_attestation_id.cpp
+++ b/keystore/keystore_attestation_id.cpp
@@ -35,8 +35,6 @@
 #include <keystore/KeyAttestationPackageInfo.h>
 #include <keystore/Signature.h>
 
-#include <hardware/keymaster_defs.h>
-
 #include <private/android_filesystem_config.h> /* for AID_SYSTEM */
 
 #include <openssl/asn1t.h>
@@ -212,10 +210,6 @@ build_attestation_application_id(const KeyAttestationApplicationId& key_attestat
             return BAD_VALUE;
         }
         std::string package_name(String8(*pinfo->package_name()).string());
-        // Prevent Google Play Services from using key attestation for SafetyNet
-        if (package_name == "com.google.android.gms") {
-            return KM_ERROR_UNIMPLEMENTED;
-        }
         std::unique_ptr<KM_ATTESTATION_PACKAGE_INFO> attestation_package_info;
         auto rc = build_attestation_package_info(*pinfo, &attestation_package_info);
         if (rc != NO_ERROR) {
-- 
2.17.1

